src/librustc_passes/lib.rs

```diff
@@ -19,12 +19,10 @@ use rustc::ty::query::Providers;
 pub mod error_codes;
 
 pub mod ast_validation;
-pub mod rvalue_promotion;
 pub mod hir_stats;
 pub mod layout_test;
 pub mod loops;
```

Comment:
Feels like this crate is becoming quite pointless.

Reply:
We could split them out to a crate for each pass. :)

Reply:
They are tiny though... a bunch of them belong in `rustc::hir` which *could* become a crate (with only the data types remaining in `rustc`).

Reply:
Whatever we do, let's please not inflate the size of the `rustc` crate. :)